curvlinops/diagonal/hutchinson.py

```diff
@@ -1,27 +1,30 @@
 """Hutchinson-style matrix diagonal estimation."""
 
-from numpy import ndarray
+from numpy import column_stack, einsum, ndarray
 from scipy.sparse.linalg import LinearOperator
 
 from curvlinops.sampling import random_vector
 
 
-class HutchinsonDiagonalEstimator:
-    r"""Class to perform diagonal estimation with Hutchinson's method.
+def hutchinson_diag(
+    A: LinearOperator, num_matvecs: int, distribution: str = "rademacher"
```

Comment:
No strong opinion, but maybe something like `num_samples` instead of `num_matvecs` is a bit more intuitive in the context of a stochastic estimator?

Reply:
I thought so too when I first implemented these methods. After reading more into the literature (Hutch++, XTrace), I think number of matrix-vector queries is better, because it applies to all trace estimators. E.g. for Hutch++, number of samples is different from number of matrix-vector products (because two thirds of the matvecs is used for deterministic trace estimation, one third for 'sampled' trace estimation). Using `num_matvecs` also allows these methods to be compared more easily (i.e. using same number of `matvecs`), because users don't need to know how these methods work in detail.

Reply:
Ok makes sense!